### ISSUE TITLE
Allow ROOT::RDF::RCutFlowReport and ROOT::RDF::TCutInfo to be written to a file.

### ISSUE THREAD

**Issue by mholtrop** [root-project/root#14815]
### Explain what you would like to see improved and how.

Currently you cannot read a ROOT::RDF::RCutFlowReport back from a file because ROOT::RDF::TCutInfo does not have a public constructor. It would be really great to be able to store the RCutFlowReport in a file together with the snapshot of the resulting RDataFrame. This would allow an easy way to keep track of the cut flow in filtered data that is stored back into ROOT files.

It would be nice if you could do (PyROOT example here):

```
import ROOT as R
df = R.RDataFrame("Data","data_file.root")
df_filtered = df.Filter("x>10","xcut").Filter("y>10","ycut")
report = df_filtered.Report()
df_filtered.SnapShot("tmp","test_file.root")
file = R.TFile("test_file.root","UPDATE")
file.WriteObject(report.GetValue(),"cut_flow")
file.Close()
```

The above code works, but the "cut_flow" in the file cannot be read back. 
I am not sure this is because TCutInfo cannot be read back from a file because of the lack of a public constructor, but it appears to be the case.

### ROOT version

ROOT 6.31/01 

### Installation method

build from source obtained from GitHub master

### Operating system

MacOS

**Comment by vepadulano** [comment]
Dear @mholtrop ,

Thanks for your report. A fix is provided in the linked PR. If you are able to build ROOT from the branch of the PR, a confirmation of it fixing your issue would be appreciated, thanks!

**Comment by mholtrop** [comment]
Nice fast response! 
I tested this from Python and it worked:
```
file = R.TFile("ele_2019_dfx_filtered.root","READ")
cut_flowx = file.Get("cut_flow")
cut_flowx.Print()
```

resulted in:
```
Select only 1 Primary MC particle: pass=2012672    all=2012917    -- eff=99.99 % cumulative eff=99.99 %
Select only 1 KF track.: pass=952195     all=2012672    -- eff=47.31 % cumulative eff=47.30 %
Select only 1 Primary MC particle at ECal: pass=216318     all=952195     -- eff=22.72 % cumulative eff=10.75 %
Select only 1 Primary MC particle at SVT6: pass=38452      all=216318     -- eff=17.78 % cumulative eff=1.91 %
Select only 1 ECal cluster: pass=34566      all=38452      -- eff=89.89 % cumulative eff=1.72 %
```

which was the expected output.
Thanks!

**Comment by vepadulano** [comment]
You are welcome! This feature will be available in the next stable release 6.32 :+1:
